include/deal.II/base/hdf5.h

```diff
@@ -0,0 +1,1073 @@
+     * dimensions is a one-dimensional array of size rank specifying the size of
+     * each dimension of the dataset.
+     */
+    std::vector<hsize_t>
+    get_dimensions() const;
+
+    /**
+     * This function returns the total number of elements in the dataset.
+     */
+    unsigned int
+    get_size() const;
+
+    /**
+     * This function returns the rank of the dataset.
+     */
+    unsigned int
+    get_rank() const;
+
+  private:
+    /**
+     * Rank of the DataSet
+     */
+    unsigned int rank;
+
+    /**
+     * The vector `dimensions` is a one-dimensional array of size rank
+     * specifying the size of each dimension of the dataset.
+     */
+    std::vector<hsize_t> dimensions;
+
+    /**
+     * HDF5 dataspace identifier.
+     */
+    std::shared_ptr<hid_t> dataspace;
+
+    /**
+     * Total number of elements in the dataset.
+     */
+    unsigned int size;
+
+    /**
+     * If query_io_mode is set to true, then after every read/write operation
+     * the HDF5 deal.ii interface calls the routines
+     * [H5Pget_mpio_actual_io_mode()](https://support.hdfgroup.org/HDF5/doc/RM/RM_H5P.html#Property-GetMpioActualIoMode)
+     * and
+     * [H5Pget_mpio_no_collective_cause()](https://support.hdfgroup.org/HDF5/doc/RM/RM_H5P.html#Property-GetMpioNoCollectiveCause).
+     * The results are stored in io_mode, local_no_collective_cause and
+     * global_no_collective_cause.
+     */
+    bool query_io_mode;
+
+    /**
+     * I/O mode that was performed on the last parallel I/O call.
+     */
+    H5D_mpio_actual_io_mode_t io_mode;
+
+    /**
+     * Local causes that broke collective I/O on the
+     * last parallel I/O call. See <a
+     * href="https://support.hdfgroup.org/HDF5/doc/RM/RM_H5P.html#Property-GetMpioNoCollectiveCause">H5Pget_mpio_no_collective_cause</a>.
+     */
+    uint32_t local_no_collective_cause;
+
+    /**
+     * Global causes that broke collective I/O on the
+     * last parallel I/O call. See <a
+     * href="https://support.hdfgroup.org/HDF5/doc/RM/RM_H5P.html#Property-GetMpioNoCollectiveCause">H5Pget_mpio_no_collective_cause</a>.
+     */
+    uint32_t global_no_collective_cause;
+  };
+
+  /**
+   * This class implements an HDF5 Group
+   *
+   * @author Daniel Garcia-Sanchez, 2018
+   */
+  class Group : public HDF5Object
+  {
+  protected:
+    /**
+     * Group access mode
+     */
+    enum class GroupAccessMode
+    {
+      /**
+       * Opens an existing group
+       */
+      open,
+      /**
+       * Creates a new group
+       */
+      create
+    };
+    /**
+     * This constructor creates or opens a group depending on the value of
+     * @p mode. The group will be placed insided the group @p parent_group. The
+     * parameter @p mpi defines if the the I/O operations are serial or
+     * parallel. This is an internal constructor, the functions open_group() and
+     * create_group() should be used to open or create a group.
```

Comment:
which class do these functions belong to?

Reply:
oh, I see, these are of the current class. Maybe say "the functions ... of the current class should be used..."